File: app/src/main/java/com/simpleshift/scheduler/domain/model/CommonRestResult.kt
package com.simpleshift.scheduler.domain.model

import androidx.compose.runtime.Immutable
import java.time.LocalDate

@Immutable
data class CommonRestResult(
    val teamAName: String,
    val teamBName: String,
    val nextCommonRestDate: LocalDate?,
    val daysUntilNext: Int?,
    val commonRestDates: List<LocalDate>,
    val totalCount: Int,
    val countIn30Days: Int,
    val countIn60Days: Int
)

File: flutter/lib/domain/models/common_rest_result.dart
/// 共同休息结果 — 对应 Android 版 CommonRestResult.kt
class CommonRestResult {
  final String teamAName;
  final String teamBName;
  final DateTime? nextCommonRestDate;
  final int? daysUntilNext;
  final List<DateTime> commonRestDates;
  final int totalCount;
  final int countIn30Days;
  final int countIn60Days;

  const CommonRestResult({
    required this.teamAName,
    required this.teamBName,
    this.nextCommonRestDate,
    this.daysUntilNext,
    required this.commonRestDates,
    required this.totalCount,
    required this.countIn30Days,
    required this.countIn60Days,
  });
}
